readthedocs/redirects/utils.py

```diff
@@ -84,7 +84,7 @@ def get_redirect_response(request, full_path):
         language, version_slug, path = language_and_version_from_path(path)
 
     path, http_status = project.redirects.get_redirect_path_with_status(
-        path=path, language=language, version_slug=version_slug
+        path=path, full_path=full_path, language=language, version_slug=version_slug
```

Comment:
I don't know why we were passing this `full_path` already. We are re-calculating it for one of the cases at `redirect_exact` method.

Passing it at this point, allows us to use it as a value to compare at db level instead from Python itself.